---
title: A generalized linear mixed model for miRNA-seq data
author: '[Andrea M. Baran and Matthew N. McCall](http://mnmccall.com)'
date: "`r Sys.Date()`"
output:
  html_document: default
  pdf_document: default
  word_document: default
---


```{r global_options, include=FALSE}
library(knitr)
knitr::opts_chunk$set(fig.width=12, fig.height=8,
                      warning=FALSE, message=FALSE)
```

# Introduction

In this vignette we demonstrate how to analyze microRNA-seq data at the isomiR-level using the miRglmm package. This package begins with a SummarizedExperiment object containing aligned isomiR-level read counts. The alignment can be performed by either miRge or srabench. We provide instructions for both alignment algorithms, starting with the fastq files and concluding with the SummarizedExperiment object to be used as input to the miRglmm function for modeling. Each step in the analysis is briefly described and carried out by one or more functions implemented in this package. Each of these functions has its own help file describing it in detail; here we focus on the analysis as a whole. 

# Experimental Design
Our example data set consists of 6 monocyte samples. Fastq files for these samples can be accessed through the sequence read archive (SRA), via this link:


# miRge processing and alignment
Arun: add miRge instructions here

# srabench processing and alignment
Ernesto: add srabench instructions here

# Statistical modeling with miRglmm

```{r load_miRglmm}
library('miRglmm')
```

```{r load_example_data}
data("se_example")
```

The example data and corresponding annotation are stored in a SummarizedExperiment object:

```{r data_explore}
show(se_example)
str(assay(se_example))
```

The colData contain information about each sample:
```{r col_data_explore}
colData(se_example)
```

The rowData contain information about each isomiR, including its sequence and the miRNA it aligned to:
```{r col_data_explore}
rowData(se_example)
```

## Fitting the GLMM model
Andrea: add instructions for miRglmm usage here

```{r miRglmm_fit}
fit = miRglmm(se_example, col_group=rep(c("A","B"), c(3,3)))
```

## Output of the GLMM model
Andrea: add info about the output of the miRglmm

## Downstream analyses 
Andrea: add some example of what you can do with the output in terms of results / plots

# Session Info
```{r session_info}
sessionInfo()
```
